dev.finances.js

```diff
@@ -1,107 +1,225 @@
 const Modal = {
-    open() {
-        document.querySelector('.modal-overlay').classList.add('active')
+  open() {
+    document.querySelector('.modal-overlay').classList.add('active')
+
+  },
+  close() {
+    document.querySelector('.modal-overlay').classList.remove('active')
+  }
+}
+
+const Storage = {
+  get() {
+    return JSON.parse(localStorage.getItem("dev.finances:transactions")) || []
+  },
+  set(transactions) {
+    localStorage.setItem("dev.finances:transactions", JSON.stringify(transactions))
+  }
+}
+
+
+const Transaction = {
+  all: Storage.get(),
+
+  add(transaction){
+    Transaction.all.push(transaction)
+
+    App.reload()
+  },
+  remove(index) {
+    Transaction.all.splice(index, 1)
+
+    App.reload()
+  },
+  // Somar as entradas
+  incomes() {
+    let income = 0;
+    Transaction.all.forEach(transaction => {
+      if(transaction.amount > 0) {
+        income += transaction.amount;
+      }
+    })
+    return income;
+  },
+  // Somar as saidas
+  expenses() {
+    let expense = 0;
+    Transaction.all.forEach(transaction => {
+      if(transaction.amount < 0) {
+        expense += transaction.amount;
+      }
+    })
+
+    return expense;
+  },
   
-    },
-    close() {
-        document.querySelector('.modal-overlay').classList.remove('active')
-    }
-     }  
-     
-     
-     const transactions = [{
-         id: 1,
-         description: 'Luz',
-         amout: -50000,
-         date: '23/01/2021'
-     },
-     {
-         id: 2,
-         description: 'Website',
-         amout: 500000,
-         date: '23/01/2021'
-     },
-     {
-         id: 3,
-         description: 'Internet',
-         amout: -20000,
-         date: '23/01/2021'
-     },
-
-     {
-        id: 4,
-        description: 'App',
-        amout: 200000,
-        date: '23/01/2021'
-
-     }
+  // Entradas - Saidas
+  total() {
     
-    ]
+    return Transaction.incomes() + Transaction.expenses();
```

Comment:
Aqui não seria com sinal de menos? Tipo entradas - saídas? hehe

```suggestion
    return Transaction.incomes() - Transaction.expenses();
```